ambry-utils/src/main/java/com.github.ambry.utils/Utils.java

```diff
@@ -890,6 +893,18 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
     }
   }
 
+  /**
+   * Set permissions for given files.
+   * @param files a list of files that set specified permissions
+   * @param permissions a set of {@link PosixFilePermission} associated with given files
+   * @throws IOException
+   */
+  public static void setFilePermission(List<File> files, Set<PosixFilePermission> permissions) throws IOException {
```

Comment:
Since most of this method's callers pass only one file, I would add a version that accepts a single file, to avoid wrapping in a collection each time.

Reply:
Right. For single file case, looks like there is no need to write another version, I can directly use `Files.setPosixFilePermissions` method.

Reply:
`setFilesPermission`?